allensdk/brain_observatory/filtered_running_speed/__main__.py

```diff
@@ -0,0 +1,954 @@
+import warnings
+
+import numpy as np
+import pandas as pd
+import scipy.signal as signal
+
+from scipy.stats import zscore
+
+from allensdk.brain_observatory import sync_utilities
+from allensdk.brain_observatory.argschema_utilities import \
+    ArgSchemaParserPlus, \
+    write_or_print_outputs
+from allensdk.brain_observatory.sync_dataset import Dataset
+from ._schemas import InputParameters, OutputParameters
+from typing import Union, Iterable, Optional
+
+from allensdk.brain_observatory.filtered_running_speed.stim_file import (
+    CamStimOnePickleStimFile,
+    BehaviorPickleFile,
+    ReplayPickleFile
+)
+
+DEGREES_TO_RADIANS = np.pi / 180.0
+INDEX_TO_BEHAVIOR = 0
+INDEX_TO_MAPPING = 1
+INDEX_TO_REPLAY = 2
+DEFAULT_SAMPLING_FREQUENCY = 60
+CRITICAL_FREQUENCY = 4
+FILTER_ORDER = 3
+DEFAULT_ZSCORE_THRESHOLD = 10.0
+USE_LOWPASS_FILTER = True
+WHEEL_DIAMETER_IN_INCHES = 6.5
+START_FRAME = 0
+
+# in cm/in
+WHEEL_DIAMETER = WHEEL_DIAMETER_IN_INCHES * 2.54
+
+
+def check_encoder(parent, key):
+    """
+    checks if encoder exists
+    ---------
+    parent: dict
+        stim file data
+    key: string
+        A key to check
+    Returns
+    -------
+    boolean
+        True if encoder exists in stim data
+    """
+    result = True
+
+    if len(parent["encoders"]) != 1:
+        result = False
+
+    elif key not in parent["encoders"][0]:
+        result = False
+
+    elif len(parent["encoders"][0][key]) == 0:
+        result = False
+
+    return result
+
+
+def calc_deriv(x, time):
+    """
+    Calculate the derivative
+    ---------
+    x: np.ndarray
+        data values
+    time: np.ndarray
+        time values
+    Returns
+    -------
+    np.ndarray
+        The derivative
+    """
+    dx = np.diff(x, prepend=np.nan)
+    dt = np.diff(time, prepend=np.nan)
+
+    return dx / dt
+
+
+def _zscore_threshold_1d(
```

Comment:
This function already exists. See https://github.com/AllenInstitute/AllenSDK/blob/master/allensdk/brain_observatory/behavior/data_objects/running_speed/running_processing.py#L277

Reply:
Thanks. I've refactored my code to call the existing function.